wolk/WolkSingle.cpp

```diff
@@ -0,0 +1,141 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "wolk/WolkSingle.h"
+
+#include "wolk/WolkBuilder.h"
+
+namespace wolkabout
+{
+namespace connect
+{
+WolkBuilder WolkSingle::newBuilder(Device device)
+{
+    return WolkBuilder(device);
+}
+
+void WolkSingle::addReading(const std::string& reference, std::string value, std::uint64_t rtc)
+{
+    if (rtc == 0)
+    {
+        rtc = WolkSingle::currentRtc();
+    }
+
+    addToCommandBuffer([=]() -> void { m_dataService->addReading(m_device.getKey(), reference, value, rtc); });
+}
+
+void WolkSingle::addReading(const std::string& reference, const std::vector<std::string>& values, std::uint64_t rtc)
+{
+    if (rtc == 0)
+    {
+        rtc = WolkSingle::currentRtc();
+    }
+
+    addToCommandBuffer([=]() -> void { m_dataService->addReading(m_device.getKey(), reference, values, rtc); });
+}
+
+void WolkSingle::addReading(const Reading& reading)
+{
+    addToCommandBuffer([this, reading] { m_dataService->addReading(m_device.getKey(), reading); });
+}
+
+void WolkSingle::addReadings(const std::vector<Reading>& readings)
+{
+    addToCommandBuffer([this, readings] { m_dataService->addReadings(m_device.getKey(), readings); });
+}
+
+void WolkSingle::registerFeed(const Feed& feed)
+{
+    addToCommandBuffer([=]() -> void { m_dataService->registerFeed(m_device.getKey(), feed); });
+}
+
+void WolkSingle::registerFeeds(const std::vector<Feed>& feeds)
+{
+    addToCommandBuffer([=]() -> void { m_dataService->registerFeeds(m_device.getKey(), feeds); });
+}
+
+void WolkSingle::removeFeed(const std::string& reference)
+{
+    addToCommandBuffer([=]() -> void { m_dataService->removeFeed(m_device.getKey(), reference); });
+}
+
+void WolkSingle::removeFeeds(const std::vector<std::string>& references)
+{
+    addToCommandBuffer([=]() -> void { m_dataService->removeFeeds(m_device.getKey(), references); });
+}
+
+void WolkSingle::pullFeedValues()
+{
+    addToCommandBuffer([=]() -> void { m_dataService->pullFeedValues(m_device.getKey()); });
+}
+
+void WolkSingle::pullParameters()
+{
+    addToCommandBuffer([=]() -> void { m_dataService->pullParameters(m_device.getKey()); });
+}
+
+void WolkSingle::addAttribute(Attribute attribute)
+{
+    addToCommandBuffer([=]() -> void { m_dataService->addAttribute(m_device.getKey(), attribute); });
+}
+
+void WolkSingle::updateParameter(Parameter parameter)
+{
+    addToCommandBuffer([=]() -> void { m_dataService->updateParameter(m_device.getKey(), parameter); });
+}
+
+std::unique_ptr<ErrorMessage> WolkSingle::awaitError(std::chrono::milliseconds timeout)
```

Comment:
Why do we need this in public api? User probably wouldn't know when to expect an error. This should probably be a callback, not a synced call